### DISCUSSION TITLE
How to make query for SQL language injection

### DISCUSSION THREAD

**Discussion by augustocdias** [tree-sitter/tree-sitter#1577] [Q&A] [answered]
I'm trying to create a query for inject SQL in raw strings in my code. In the playground I started to have some positive results using something on those lines:

```
((
  (raw_string_literal) @constant
  (#match? @constant "(SELECT|select|insert|INSERT).*")
)@injection.content (#set! injection.language "sql"))
```

My question is about the `match`. Is it possible to use the content regex of another language for that? Tree-sitter already can detect SQL language, so I was wondering if I could somehow reuse this content matcher for this instead of writing a big regex that can match every possibility.

**Comment by maxbrunsfeld** [comment, marked as answer]
I think the way you're doing it is good. Tree-sitter doesn't have any existing way to check an arbitrary block of text and determine if it is or is not SQL. So just looking for `select`/`insert`/`update`/`delete`/`create table`/`alter table` is probably the most practical approach.
